### PR TITLE
vim-patch:8.2.1535: it is not possible to specify cell widths of characters

### PR DESCRIPTION
**Pull request by delphinus**
Problem:    It is not possible to specify cell widths of characters.
Solution:   Add setcellwidths().
https://github.com/vim/vim/commit/08aac3c6192f0103cb87e280270a32b50e653be1

### CODE REVIEW

**Comment by delphinus** [comment]
TODO: Later I'll commit for `v8.2.1537: memory acccess error when using setcellwidths()` here.

File: src/nvim/eval/funcs.c [hunk 1]
```diff
@@ -8175,6 +8176,116 @@ static void f_setbufvar(typval_T *argvars, typval_T *rettv, FunPtr fptr)
...
 
+static int tv_nr_compare(const void *a1, const void *a2)
+{
+  listitem_T *li1 = (listitem_T *)a1;
+  listitem_T *li2 = (listitem_T *)a2;
+
+  return li1->li_tv.vval.v_number - li2->li_tv.vval.v_number;
+}
+
+// "setcellwidths()" function
```
Comment (bfredl) [inline src/nvim/eval/funcs.c:8187]: three slashes

File: src/nvim/mbyte.c [hunk 2]
```diff
@@ -458,6 +458,55 @@ static bool intable(const struct interval *table, size_t n_items, int c)
   return false;
 }
 
+cw_interval_T *cw_table = NULL;
+size_t cw_table_size = 0;
+
+// Return 1 or 2 when "c" is in the cellwidth table.
```
Comment (bfredl) [inline src/nvim/mbyte.c:464]: three slashes and use `@return`

File: src/nvim/mbyte.c [hunk 3]
```diff
@@ -472,12 +521,31 @@ static bool intable(const struct interval *table, size_t n_items, int c)
...
+    //
+    n = wcwidth(c);
+
+    if (n < 0) {
+      return 6;                 // unprintable, displays <xxxx>
+    }
+    if (n > 1) {
+      return n;
+    }
+#else
```
Comment (bfredl) [inline src/nvim/mbyte.c:541]: this should be deleted

File: src/nvim/eval/funcs.c (hunk 1 above)
Comment (delphinus) [reply src/nvim/eval/funcs.c]: Fixed on f942ce6

File: src/nvim/mbyte.c (hunk 2 above)
Comment (delphinus) [reply src/nvim/mbyte.c]: Fixed on f37505b

File: src/nvim/mbyte.c (hunk 3 above)
Comment (delphinus) [reply src/nvim/mbyte.c]: Fixed on 7a2620506

File: src/nvim/eval/funcs.c [hunk 4]
```diff
@@ -8132,6 +8132,156 @@ static void set_position(typval_T *argvars, typval_T *rettv, bool charpos)
...
+  cw_table_save = cw_table;
+  cw_table_size_save = cw_table_size;
+  cw_table = table;
+  cw_table_size = (size_t)l->lv_len;
+
+  // Check that the new value does not conflict with 'fillchars' or 'listchars'.
+  if (set_chars_option(curwin, &p_fcs, false) != NULL) {
+    error = "E835: Conflicts with value of 'fillchars'";
+  } else if (set_chars_option(curwin, &p_lcs, false) != NULL) {
+    error = "E835: Conflicts with value of 'listchars'";
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:8261]:
```suggestion
    error = "E834: Conflicts with value of 'listchars'";
```

File: src/nvim/eval/funcs.c [hunk 5]
```diff
@@ -8132,6 +8132,156 @@ static void set_position(typval_T *argvars, typval_T *rettv, bool charpos)
...
+  } else {
+    FOR_ALL_TAB_WINDOWS(tp, wp) {
+      if (set_chars_option(wp, &wp->w_p_lcs, false) != NULL) {
+        error = "E834: Conflicts with value of 'listchars'";
+        break;
+      }
+      if (set_chars_option(wp, &wp->w_p_fcs, false) != NULL) {
+        error = "E834: Conflicts with value of 'fillchars'";
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:8269]:
```suggestion
        error = "E835: Conflicts with value of 'fillchars'";
```

**Comment by zeertzjq** [comment]
This is too hard to rebase. Restarted in #19674

**Pull request closed without merging**